### ISSUE TITLE
ilab taxonomy diff doesn't use taxonomy specified in init

### ISSUE THREAD

**Issue by danmcp** [instructlab/instructlab#2086]
**Describe the bug**
ilab taxonomy diff doesn't use repo specified with init

**To Reproduce**
1. ilab config init --taxonomy-path </home/user>/taxonomy
2. ilab taxonomy diff

and you get:

$ ilab taxonomy diff
Reading taxonomy failed with the following error: /home/user/.local/share/instructlab/taxonomy

**Expected behavior**
The repo I specified on init is used for the diff

**Comment by danmcp** [comment]
Looks like there is an attempt to use generate's config:

```
    # load defaults from config ctx obj if not specified via CLI arg
    if not taxonomy_base:
        taxonomy_base = ctx.obj.config.generate.taxonomy_base
    if not taxonomy_path:
        taxonomy_path = ctx.obj.config.generate.taxonomy_path
```

But the defaults on the options are winning

**Comment by bjhargrave** [comment]
See my fix in #1848.
